src/test/compressor/test_compression.cc

```diff
@@ -480,3 +480,78 @@ TEST(ZlibCompressor, isal_compress_zlib_decompress_walk)
 }
 
 #endif	// __x86_64__
+
+#ifdef HAVE_QATZIP
+TEST(QAT, enc_qat_dec_noqat) {
+#ifdef HAVE_LZ4
+  const char* alg_collection[] = {"zlib", "lz4", "snappy"}; 
+#else
+  const char* alg_collection[] = {"zlib", "snappy"}; 
+#endif
+  for (auto alg : alg_collection) {
+    g_conf->set_val("qat_compressor_enabled", "true");
+    CompressorRef q = Compressor::create(g_ceph_context, alg);
```

Comment:
Is there an assert we can make here to be sure we're constructing the right kind of compressor, with the qat flag appropriately set?  if the q version doesn't' have it set we can exit out with a cout << "warning, failed to construct qat compressor, skipping test" message.  I'm just worried about a false positive we don't notice.